plugins/package-managers/python/src/main/kotlin/Poetry.kt

```diff
@@ -143,3 +157,37 @@ internal fun parseScopeNamesFromPyproject(pyprojectFile: File): Set<String> {
 
     return scopes
 }
+
+internal fun getPythonVersion(constraint: String): String? {
+    val rangeLists = constraint.split(",")
+        .map { RangesListFactory.create(it) }
+        .takeIf { it.isNotEmpty() } ?: return null
+
+    return PYTHON_VERSIONS.lastOrNull { version ->
+        rangeLists.all { rangeList ->
+            val semver = Semver.coerce(version)
+            semver != null && rangeList.isSatisfiedBy(semver)
+        }
+    }
+}
+
+internal fun getPythonVersionConstraint(pyprojectTomlFile: File): String? {
+    val dependenciesSection = getTomlSectionContent(pyprojectTomlFile, "tool.poetry.dependencies")
+        ?: return null
+
+    return dependenciesSection
+        .split("\n")
+        .map { it.trim() }
+        .find { it.startsWith("python = ") }
+        ?.let { it.removePrefix("python = ") }?.removeSurrounding("\"")
```

Comment:
I believe these 3 lines could be simplified to

        .firstNotNullOfOrNull { it.trim().withoutPrefix("python = ") }
        ?.removeSurrounding("\"")